src/backend/replication/logical/message.c

```diff
@@ -93,3 +97,86 @@ logicalmsg_redo(XLogReaderState *record)
 
 	/* This is only interesting for logical decoding, see decode.c. */
 }
+
+/*
+ * NEON: remove AUX object
+ */
+void
+wallog_file_removal(char const* path)
+{
+	char	prefix[MAXPGPATH];
+
+	/* Do not wallog AUX file at replica */
+	if (!XLogInsertAllowed())
+		return;
+
+	snprintf(prefix, sizeof(prefix), "neon-file:%s", path);
+	elog(DEBUG1, "neon: deleting contents of file %s", path);
+
+	/* unlink file */
+	LogLogicalMessage(prefix, NULL, 0, false, true);
+}
+
+/*
+ * NEON: persist file in WAL to save it in persistent storage.
+ */
+void
+wallog_file_descriptor(char const* path, int fd, uint64_t limit)
```

Comment:
Because this uses `palloc`, this will always fail with a file larger than MaxAllocSize. IIRC that's the max size of a record that you can WAL-log anyway, or maybe a little less. I think that 's fine, but maybe just add a comment about it.

Reply:
Done